internal/rootcoord/timeticksync.go

```diff
@@ -101,20 +169,45 @@ func (t *timetickSync) UpdateTimeTick(in *internalpb.ChannelTimeTickMsg) error {
 	if !ok {
 		return fmt.Errorf("Skip ChannelTimeTickMsg from un-recognized proxy node %d", in.Base.SourceID)
 	}
+
+	// if from ddl operation, remove ddl timetick firstly
+	if ddl {
+		t.RemoveDdlTimeTick(in.DefaultTimestamp, reason)
```

Comment:
t.RemoveDdlTimeTick is called to remove timestamp, I would suggest to call t.RemoveDdlTimeTick before t.core.SendTimeTick(ts, true, reason) and remote ddl parameter

Reply:
sorry ,remove ddl parameter, this also help solve the deadlock

Reply:
updated